icu4c/source/i18n/unicode/usearch.h

```diff
@@ -280,9 +280,15 @@ typedef enum {
 /* open and close ------------------------------------------------------ */
 
 /**
-* Creating a search iterator data struct using the argument locale language
+* Create a String Search iterator data struct using the argument locale language
 * rule set. A collator will be created in the process, which will be owned by
-* this search and will be deleted in <tt>usearch_close</tt>.
+* this String Search and will be deleted in <tt>usearch_close</tt>.
+*
+* The UStringSearch retains a pointer to both the pattern and text strings.
```

Comment:
This might be true... but is it actually true about the pattern string?

Reply:
Fair question. :)

I created a simple/hacky test program to double-check here:
https://gist.github.com/jefgen/261424866469a1ad3ec6c35dd16a72e6

The short summary is:
- For the `pattern` input string, if the caller deletes the string then the UStringSearch object will still continue to "work" and give valid results.
- However, for the `text` input string, if the caller deletes the string then the UStringSearch object won't crash -- but it will start giving incorrect or invalid results (as it will be searching the deleted memory).
- That said, for both of them:
If the caller deletes/frees the strings then the two APIs `usearch_getPattern` and `usearch_getText` will give back a pointer to the free'd strings as the result.

Although, perhaps the last one might be expected or even perhaps acceptable maybe...?
But on the other hand, if the caller assumed that the `usearch_open*` API(s) made copies of the input strings, then they might expect to still get back a valid string(s).

Reply:
Thanks for checking. Let's play it safe.